### ISSUE TITLE
New Refined Bookmarklet/Plugin-UI

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/backend#226]
Again, the designer has given suggestions on our dev Plugin-UI. We can discuss which of these suggestions we can/can't/will/won't implement. 

![button ux](https://cloud.githubusercontent.com/assets/4953703/18412125/224b8862-777f-11e6-871e-e1be5fea5ad8.png)

Some notes on this. He suggests not having a sign in page at all, and instead simple having a user put their email address in on the submit page. 
* If they're logged in, this would be pre-filled. 
* If they weren't logged in it would be empty and invite them to fill it. 
 * If account: all good
 * If no account, we give them notice they need to finish making one before we can send it.

Again, we have design files for each part of this.

**Comment by markmacgillivray** [comment]
This is pretty much what we already have in the dev plugin anyway. We already don't have a sign in page - if they have an account they are signed in, if they don't they can sign in on the site with exactly that, just one email. There is little point putting that in the plugin, because we HAVE to take them to the site anyway to get their API key into their plugin the first time round after they sign up. Just need the icons to finish off the plugin.

**Comment by JosephMcArthur** [comment]
Icons are now up in this repo. The design of the button generally here is very nice, and it would be good to implement that. 

Hard to see on the test site atm, but does the article come up at the top? That is a pretty good suggestion I think we should take. Same with the removed of "file a bug", better to just do context dependent links like "but i see one right there". 

Here is also a suggestion for what to do on blacklist pages "what does this app do again?"

The thing we didn't get to here was a design for bigger errors from the backend and things like that.

**Comment by JosephMcArthur** [comment]
Update here with a suggested page for when the backend throws errors. 

![button ux](https://cloud.githubusercontent.com/assets/4953703/18549644/adbc3204-7b47-11e6-886f-676309248868.png)

**Comment by chealsye** [comment]
Really liking these!

**Comment by JosephMcArthur** [comment]
Commented on this now! 

![button_ux_commented](https://cloud.githubusercontent.com/assets/4953703/19112197/71bfd80a-8afb-11e6-9eda-9eb90858ab3b.png)

**Comment by chealsye** [comment]
@mark do you have an example of a bug form?

@mc this is labeled as blocked: copy. Is there anything else you need from me?



Pre-filled social media messages
Success
T: The @OA_Button got me access to the research I needed! Need articles or data? Create your own request: [link]

FB: The Open Access Button got me access to the research I needed! [link] 

Need articles or data? Create your own request at openaccessbutton.org.


Request:
T: I need access to research. Do you, too? Support my @OA_Button request! [link]
FB: I need access to research. Do you, too? Support my Open Access Button request, please! [link]

You can create your own requests for articles and data at openaccessbutton.org.


We can tweak or revisit these, of course.

**Comment by mc** [comment]
Not sure, i hope the email adress is being atleast verified?

**Comment by markmacgillivray** [comment]
@chealsye no, I don't have an example of a bug form. Also I think you have tagged the wrong people in your comment.

**Comment by JosephMcArthur** [comment]
@mc yes, the email is being verified! Thanks for actually responding to the tag :) 

@markmacgillivray / @chealsye Mark, we've spoken about setting up a bug form before - I think it would be nice although that's all it is. I'll make an issue for it, and for now sending people to repo's or bugs@openaccessbutton.org will be a-okay.

**Comment by markmacgillivray** [comment]
Unfortunately I think some of the new stuff here may not be usable. We cannot overlay the word count onto the textbox, because the user has to be able to see in it and type in it. But we no longer have an email box, and the big bottom submit box works well as an indicator of when submission can be made, and looks good I think.

I can put in the share success and share request links though, will do that.

The text to go to a bug form could work, and either be "but i see one right there" or "not the right article?".

I can see the article and data icons in the Assets folder of this repo, but they are green. Can we have them in black? That way I can just use the same ones overlaid on different colours behind, for green success / orange failure.

However I cannot find the orange loading spinner or the broken cogs image - @JosephMcArthur do you know where they are? If so I can use them. If not, I have a simple loading spinner I can use instead, but not an error icon.

**Comment by JosephMcArthur** [comment]
> Unfortunately I think some of the new stuff here may not be usable. We cannot overlay the word count onto the textbox, because the user has to be able to see in it and type in it. But we no longer have an email box, and the big bottom submit box works well as an indicator of when submission can be made, and looks good I think.

Okay, the email box & prompt not being present is a-okay. 

> I can put in the share success and share request links though, will do that.

@chealsye has copy above for the text in these [here](https://github.com/OAButton/backend/issues/226#issuecomment-252813089) which is good for now. Although, I don't think we have text for the prompts (other than what the designer made). 

> The text to go to a bug form could work, and either be "but i see one right there" or "not the right article?".

That text was different because it was coming up in different cases. The 2nd is just always there, incase we don't get the right metadata. The 1st was for when we think we haven't found an article. In both cases, the ideal thing to happen would be for a DOI or Article title box to come up for manual entry which then allows for searching via the find system, but it should at the least should go to a bug form. 

> I can see the article and data icons in the Assets folder of this repo, but they are green. Can we have them in black? That way I can just use the same ones overlaid on different colours behind, for green success / orange failure.

The ones I see are orange (typo?). Can you put them in for now, and @chealsye should be able to make black ones from the design files (eps / ai) - am I right? 

> However I cannot find the orange loading spinner or the broken cogs image - @JosephMcArthur do you know where they are? If so I can use them. If not, I have a simple loading spinner I can use instead, but not an error icon.

They're indeed not in the assets file. However, they are included in the design files for the UI [here](https://github.com/OAButton/backend/blob/master/UX/Button%20UX.ai). @chealsye, again, can you try and extract these elements? 

As an aside, @markmacgillivray, should we actually have these assets in the unified extension repo? Can't imagine why I thought they should go here. Seems smarter to do that now rather than fix later.

**Comment by markmacgillivray** [comment]
The user, on the plugin, will never see the metadata that we try to extract from the page. So "not the right article" has no purpose there, if that is what you meant it to be for. In which case, I will just not put it on the plugin.

Similarly in the case of the first one "but I see one right there" - there is no functionality in the plugin to extract data from the page, or to receive it from the user. This is the point of sending them to the request page, which is now editable, and where we can add/change things as we see fit, without having to release new plugins all the time, and get stuck in this annoying plugin dev release/accept cycle. So from this one I can link the user to a bug form where they can tell us the URL they were on, and we or they can then manually create a new request on the site anyway - again keeping functionality out of the plugin, in favour of having it on the site where we can change it whenever we want and also where we can point bookmarklet and other users to.

If we don't have the icons in black, and can't get them from the designer, then I may as well recolour them myself. Then I can use suitable ones when I am adding them to the plugin. Similarly for the broken cogs. The orange spinner is harder because it is overlaid on background content. I can use the spinner I already have for that though.

Don't put all the assets in the unified-extension repo - I am already copying over just the ones I need when I need them, and it would be better not to fill that repo up with stuff we don't actually use, so just keep them where they are, in the backend repo.

**Comment by JosephMcArthur** [comment]
> The user, on the plugin, will never see the metadata that we try to extract from the page. 

You're talking about not even displaying the title of the article the users is on here, yes? So a user can visually confirm that the plugin has found & is searching for the right thing. We know that from the backend alone this can go wrong, and it doesn't seem like this will be infrequent enough for this not to be an issue (correct me if I'm wrong). 

> So "not the right article" has no purpose there, if that is what you meant it to be for. In which case, I will just not put it on the plugin.

It has a clear purpose, to allow a user to flag when the system is entirely off target. However, if they can't see that then no, there isn't much point in having this. 

> Similarly in the case of the first one "but I see one right there" - there is no functionality in the plugin to extract data from the page, or to receive it from the user. This is the point of sending them to the request page, which is now editable, and where we can add/change things as we see fit, without having to release new plugins all the time, and get stuck in this annoying plugin dev release/accept cycle. So from this one I can link the user to a bug form where they can tell us the URL they were on, and we or they can then manually create a new request on the site anyway - again keeping functionality out of the plugin, in favour of having it on the site where we can change it whenever we want and also where we can point bookmarklet and other users to.

So, "but I see one right there" wouldn't have much use if the user can't visually confirm the plugin has found an article. 

As you know, I'm all for removing ourselves from this cycle & have signed off on lots of things coming out the plugin (including things I think are useful like the email) in the name of simplicity. That said, if a user fails to validate the plugin is searching for the right thing we find situations like:
1) the find system not finding the article they want, then, 
2) the request system being pointed at articles that are available
3) the user sharing requests that have massive errors

I'm just making a quick case here, as I may have misunderstood. I'm also 100% falling into the "this is how we've always done things" trap, but I also dont want to question big changes.  

> If we don't have the icons in black, and can't get them from the designer, then I may as well recolour them myself. Then I can use suitable ones when I am adding them to the plugin. Similarly for the broken cogs. 

We can do it, just might take a couple of days to get from the designer if @chealsye can't do it over the weekend. 

> The orange spinner is harder because it is overlaid on background content. I can use the spinner I already have for that though.

If it's easier, then lets do the spinner we have for now. 

> Don't put all the assets in the unified-extension repo - I am already copying over just the ones I need when I need them, and it would be better not to fill that repo up with stuff we don't actually use, so just keep them where they are, in the backend repo.

Understood, happy with this for now.

**Comment by markmacgillivray** [comment]
Hmm, yes, we would not want the user to share the request if it is wrong.
And actually, we possibly do not want users to share requests that are
still to be moderated by OAB team anyway, do we?

It probably makes sense not even to have the share buttons on the plugin.
When the plugin creates a request it gives the user a link and it also
opens the request, and we prompt them to go and look at their new request.
That is where they can fix things that could be wrong, and probably then is
when we want to prompt them to share it. So I will just add the share
buttons to the request page, where there is already a space for them to go
anyway.

Because the plugin already opens the request page for them, if there are
errors, the user is already going to have the page they are looking at
open, and also the request page, so it would be straightforward to edit in
the request page whilst reading things like the title from the article
page. This works better than the previous issues, where we had users trying
to scroll the page and that causing the plugin to disappear.

I think if we want the user to be able to tell us an article is wrong, then
that should go on the request page too - when a request is new and still in
moderation, when the originating user is viewing it, I can add a "not the
article I meant" link to the request page.

**Comment by markmacgillivray** [comment]
I found a loading spinner in the icons/PNG folder actually, I'll use that too.

**Comment by JosephMcArthur** [comment]
> Hmm, yes, we would not want the user to share the request if it is wrong.
And actually, we possibly do not want users to share requests that are
still to be moderated by OAB team anyway, do we?

The moderation is primarily for messages to authors. There is certainly an assumption here that folks who share & have that level of engagement tend not to be the ones who don't get how to use this & leave bad messages. I'm yet too this assumption not help up. 

> It probably makes sense not even to have the share buttons on the plugin.

Except in a successful case.

> When the plugin creates a request it gives the user a link and it also
opens the request, and we prompt them to go and look at their new request.
That is where they can fix things that could be wrong, and probably then is
when we want to prompt them to share it. So I will just add the share
buttons to the request page, where there is already a space for them to go
anyway.
Because the plugin already opens the request page for them, if there are
errors, the user is already going to have the page they are looking at
open, and also the request page, so it would be straightforward to edit in
the request page whilst reading things like the title from the article
page. This works better than the previous issues, where we had users trying
to scroll the page and that causing the plugin to disappear.
I think if we want the user to be able to tell us an article is wrong, then
that should go on the request page too - when a request is new and still in
moderation, when the originating user is viewing it, I can add a "not the
article I meant" link to the request page.

Okay, I'm willing to take a leap of faith here (again) and suggest for now we don't worry about having the title in & addressing issues as you suggest above. However, I will suggest that we have @meganwaps pay special addition to this in user testing. 

I also think we (I) haven't paid enough attention to the landing section of a request page as an originating user will see it. They need to have clear calls to action to check their request & change things & share. All this is relevant to #276

**Comment by markmacgillivray** [comment]
The plugin has been updated with icons now, I coloured the orange ones black  so don't need those redoing. I could not get the icon loading image to work, so used my own spinning loading one that I already have. Still doing a bit more to the plugin before making a new release, but nearly there, then you can all check the content etc.

**Comment by chealsye** [comment]
![cogs](https://cloud.githubusercontent.com/assets/9299248/19414738/a9640a3e-9324-11e6-8304-64b4c26a9f7c.png)
![spinner](https://cloud.githubusercontent.com/assets/9299248/19414739/a9684220-9324-11e6-8b9d-1f3b2de3f68d.png)

Here's the spinner and the cogs. If you need something else out of that design (or a different size) just let me know. Pulling them out just takes a minute. I couldn't find the path to the icon folder or I would have put them there.

**Comment by markmacgillivray** [comment]
Can this one be closed now? I don't think there is anything else for me to do, so unassigning for the time being.

**Comment by JosephMcArthur** [comment]
Can't close yet, we need to see / test all the UI's.

**Comment by JosephMcArthur** [comment]
@markmacgillivray can you send over a screenshot of the blacklist page, or send over the URL suffix that will evoke it?

**Comment by markmacgillivray** [comment]
Altering the plugin and plugin README now to provide a way to see this.

**Comment by JosephMcArthur** [comment]
It mentions a master branched needed to test this, but that branch doesn't exist.

**Comment by markmacgillivray** [comment]
What mentions a master branch?

**Comment by markmacgillivray** [comment]
Ah, no, it says what the defaults for each variable will be depending on whether on master or develop branch. Right now there is no master branch, but there will be as soon as we are happy to make the real release version - I will push that into a master branch.

It doesn't say how to push those variables to the plugin though, I will add that now. It was there before but I must have removed it when I backed this out that time it clashed with Megan doing testing.

**Comment by markmacgillivray** [comment]
Updated the README

**Comment by JosephMcArthur** [comment]
@markmacgillivray need screenshot

**Comment by markmacgillivray** [comment]
The error screenshots would all look as the screenshot you already added to #319 - the only difference is the text. Here are the current texts:


BLACKLISTED (THE WORD instructions IS A LINK TO /instructions):
Sorry, the page you are on is not one that we can check availability for. See the instructions for help.

UNAUTHORISED:
Unauthorised - check your API key is valid. Go to ACCOUNT_URL  and sign up if you have not already done so. Once you are signed in the plugin should find your API key for you.

FORBIDDEN:
Forbidden - please file a bug.

UNKNOWN ERROR:
Sorry, unknown error, perhaps the system is offline, or you are offline. Please file a bug including this code: ERROR_CODE

**Comment by JosephMcArthur** [comment]
Have moved the above over to the copy issue. Now closing this.
